src/gwt/src/org/rstudio/studio/client/workbench/views/tutorial/TutorialPane.java

```diff
@@ -382,6 +382,61 @@ public HandlerRegistration addTutorialNavigateHandler(Handler handler)
       return frame_.addHandler(handler, TutorialNavigateEvent.TYPE);
    }
 
+   private void updateShiny()
+   {
+      new ImmediatelyInvokedFunctionExpression()
+      {
+         private HandlerRegistration handler_;
+         private ProgressIndicator progress_;
+
+         private final String errorCaption = constants_.errorInstallingShiny();
+         private final String errorMessage =
+               constants_.errorInstallingShinyMessage();
+
+         @Override
+         protected void invoke()
+         {
+            // double-check that we were able to successfully install shiny
+            progress_ = globalDisplay_.getProgressIndicator(errorCaption);
+            handler_ = events_.addHandler(ConsolePromptEvent.TYPE, new ConsolePromptEvent.Handler()
+            {
+               @Override
+               public void onConsolePrompt(ConsolePromptEvent event)
+               {
+                  handler_.removeHandler();
+
+                  server_.isPackageInstalled("shiny", "1.6.0", new ServerRequestCallback<Boolean>()
+                  {
+                     @Override
+                     public void onResponseReceived(Boolean installed)
+                     {
+                        if (!installed)
+                        {
+                           progress_.onError(errorMessage);
+                           return;
+                        }
+
+                        progress_.onCompleted();
+                     }
+
+                     @Override
+                     public void onError(ServerError error)
+                     {
+                        Debug.logError(error);
+                        progress_.onError(errorMessage);
+                     }
+                  });
+               }
+            });
+
+            // fire console event installing learnr
+            progress_.onProgress(constants_.installingShinyCaption());
+            SendToConsoleEvent event = new SendToConsoleEvent("install.packages(\"shiny\")", true);
```

Comment:
One nit: if the user is running with an older version of R (or is using an old MRAN / RSPM checkpoint) then it's possible that shiny 1.6.0 isn't even available for them. Not sure if this is worth considering, though.

Reply:
Maybe in that case the error message should be more specific than: 

```java
    /**
     * Translated "RStudio was unable to install the shiny package.".
     *
     * @return translated "RStudio was unable to install the shiny package."
     */
    @DefaultMessage("RStudio was unable to install the shiny package.")
    @Key("errorInstallingShinyMessage")
    String errorInstallingShinyMessage();
```

Reply:
But then what happens presumably is that the tutorials pane refreshes with the same invitation to install shiny. We could check that a sufficient version of `shiny` is installable, i.e. 

```r
> .rs.availablePackages()$value["shiny", "Version"]
[1] "1.7.1"
```

As @kevinushey said though, I'm not sure it's worth it.